feat: add session-id support to ark query command [ARKQB-490]

lucaromagnoli opened the pull request:
## Checklist

- [ ] Follow the [contributor guide](./CONTRIBUTING.md)
- [ ] End-to-end tests pass
- [ ] Unit tests for essential parts of your code, e.g. APIs exposed via SDKs or endpoints
- [ ] Update `./docs`
- [ ] Recognize contributors with "@all-contributors please add <person> for <code, bug, docs, etc>"
- [ ] Linked issues #eg101

lucaromagnoli committed:
feat: add session-id support to ark query command

lucaromagnoli committed:
test: add tests for OpenAI endpoint session-id support

lucaromagnoli committed:
test: fix OpenAI endpoint test assertions to handle spec object structure

lucaromagnoli committed:
test: simplify OpenAI test assertions to use to_dict() method

lucaromagnoli committed:
test: access QueryV1alpha1Spec attributes directly instead of using to_dict()

lucaromagnoli committed:
test: use getattr for all sessionId attribute accesses in OpenAI tests

lucaromagnoli committed:
fix: update OpenAI endpoint tests to use proper ChatCompletion objects

lucaromagnoli committed:
test: add coverage for sessionId functionality in ark-cli

lucaromagnoli committed:
fix: correct ChatClient test to use mock ArkApiClient

lucaromagnoli committed:
Merge branch 'main' into feat/ark-cli-session-id

lucaromagnoli committed:
fix: add type annotation to fix TypeScript error in test

lucaromagnoli committed:
fix: compare parsed queryAnnotations objects instead of JSON strings

lucaromagnoli committed:
Merge branch 'main' into feat/ark-cli-session-id

lucaromagnoli committed:
ci: use larger runner for e2e-quickstart to fix disk pressure

danielemarostica commented on tools/ark-cli/src/lib/chatClient.ts:
```diff
@@ -61,12 +62,17 @@ export class ChatClient {
+      queryAnnotations['ark.mckinsey.com/a2a-context-id'] = config.a2aContextId;
+    }
+    if (config.sessionId) {
+      queryAnnotations['sessionId'] = config.sessionId;
```
To enable streaming we are using an annotation from constants. Do we have any constant for these annotations as well? I suggest creating these in case these are missing

lucaromagnoli committed:
ci: revert to standard runner and apply disk pressure optimizations

- Revert e2e-quickstart from gh-larger-linux-mini to ubuntu-24.04
- Remove needs dependencies to run independently (reduces wait time)
- Disable mock-llm to reduce resource usage
- Remove conditional checks for setup-container-cache-registry

lucaromagnoli committed:
ci: remove optimizations from e2e-quickstart, restore original config

lucaromagnoli committed:
refactor: add constants for query annotation keys

havasik commented on tools/ark-cli/src/lib/constants.ts:
```diff
@@ -0,0 +1,12 @@
+  A2A_CONTEXT_ID: `${ARK_PREFIX}a2a-context-id`,
+  
+  // Session ID annotation (not prefixed, used directly in queryAnnotations)
+  SESSION_ID: 'sessionId',
```
why use annotation if it'll end up in spec anyways?

havasik reviewed (changes requested):
would it make sense to also have an environment variable as well to control default session id for ark query commands?

lucaromagnoli committed:
Merge branch 'main' into feat/ark-cli-session-id

lucaromagnoli committed:
refactor: simplify sessionId handling and add env var support

- Pass sessionId directly in metadata instead of queryAnnotations (goes to spec)
- Keep queryAnnotations only for annotations (e.g., a2aContextId)
- Add ARK_SESSION_ID environment variable support for default session ID
- Update tests to reflect new structure
- Remove SESSION_ID from QUERY_ANNOTATIONS constants

lucaromagnoli committed:
chore: add vibe_artifacts directory for personal/experimental files

- Add vibe_artifacts/ to .gitignore
- Add rule to .cursorrules requiring personal files in vibe_artifacts/
- Move QUERY_DEEP_DIVE.md and a2a-streaming-error-test to vibe_artifacts/

lucaromagnoli committed:
Merge branch 'main' into feat/ark-cli-session-id

havasik commented on tools/ark-cli/src/lib/types.ts:
```diff
@@ -153,6 +153,8 @@ export interface Query {
     input: string;
     targets: QueryTarget[];
+    sessionId?: string;
+    timeout?: string;
```
Is this needed here?

lucaromagnoli replied:
Yes they are fields of the QuerySpec CRD
